python/ray/tests/test_streaming_generator_3.py

```diff
@@ -231,6 +238,93 @@ def g(sleep_time):
     assert result[10] == 4
 
 
+def test_streaming_generator_load(shutdown_only):
```

Comment:
is it deterministic to fail without this change? I was a little concern we have to send huge mount of traffic to trigger this issue.

Reply:
Every time I ran this test locally w/o the change it segfaulted. W/ the change it never did.

Reply:
what do you think we move this test to serve unit test? @edoakes @rkooo567

Reply:
if there is no pure core repro.